secure-api-gateway-ig-extensions/src/main/java/com/forgerock/sapi/gateway/fapi/v1/authorize/FapiAuthorizeRequestValidationFilter.java

```diff
@@ -69,6 +80,33 @@ protected Promise<String, NeverThrowsException> getParamFromRequest(Request requ
         return Promises.newResultPromise(getParamFromRequestQuery(request, paramName));
     }
 
+    /**
+     * Implementation which removes parameter values that don't match an entry in paramNamesToKeep from the HTTP
+     * Request's Query Parameters
+     * @param request the request from which to remove the HTTP Request's query parameters
+     * @param paramNamesToKeep the list of HTTP Request parameters to keep
+     */
+    @Override
+    protected Promise<List<String>, NeverThrowsException> removeNonMatchingParamsFromRequest(Request request, List<String> paramNamesToKeep) {
+        final Form existingQueryParams = request.getQueryParams();
+        List<String> namesToRemove = new ArrayList<>();
+        existingQueryParams.keySet().forEach((paramName)->{
+            if ( !paramNamesToKeep.contains(paramName)){
+                namesToRemove.add(paramName);
+            }
+        });
+
+        namesToRemove.forEach((paramToRemove) ->{
+            List<String> response = existingQueryParams.remove(paramToRemove);
+            if(response != null){
+                logger.debug("Removed http request parameter {}", paramToRemove);
+            }
+        });
```

Comment:
Use the keySet's iterator and Iterator.remove , then we don't need to create an additional list and can loop through once.

Reply:
This is removing entries from the Form not from a List. Wrote it this way to avoid any potential ConcurrentModificationExceptions. I don't know if removing entries while iterating over the Form (a MultiValueMap) is safe and won't result in ConcurrentModificationExceptions?

Reply:
If you use the Iterator object it is fine:
```
Iterator<String> iterator = existingQueryParams.keySet().iterator();
while (iterator.hasNext()) {
    String formParamKey = iterator.next();
    if (!paramNamesToKeep.contains(formParamKey)) {
        iterator.remove();
        logger.debug("Removed form parameter '{}' from PAR request", formParamKey);
    }
}
```